src/Ide/Plugin/Eval.hs

```diff
@@ -0,0 +1,338 @@
+
+{- Note: GhcSessionDeps
+
+Depending on GhcSession means we do need to reload all the module
+dependencies in the GHC session(from interface files, hopefully).
+
+The GhcSessionDeps dependency would allow us to reuse a GHC session preloaded
+with all the dependencies. Unfortunately, the ModSummary objects that
+GhcSessionDeps puts in the GHC session are not suitable for reuse since they
+clear out the timestamps; this is done to avoid internal ghcide bugs and
+can probably be relaxed so that plugins like Eval can reuse them. Once that's
+done, we want to switch back to GhcSessionDeps:
+
+-- https://github.com/digital-asset/ghcide/pull/694
+
+ -}
+  session <-
+    liftIO $
+      runAction "runEvalCmd.ghcSession" state $
+        use_ GhcSession $ -- See the note on GhcSessionDeps
+          toNormalizedFilePath' $
+            fp
+
+  ms <-
+    liftIO $
+      runAction "runEvalCmd.getModSummary" state $
+        use_ GetModSummary $
+          toNormalizedFilePath' $
+            fp
+
+  now <- liftIO getCurrentTime
+
+  let tmp = withSystemTempFile (takeFileName fp)
+
+  tmp $ \temp _h -> tmp $ \tempLog hLog -> do
+    liftIO $ hClose _h
+    let modName = moduleName $ ms_mod ms
+        thisModuleTarget = Target (TargetFile fp Nothing) False (Just (textToStringBuffer text, now))
+
+    hscEnv' <- ExceptT $
+      evalGhcEnv (hscEnv session) $ do
+        df <- getSessionDynFlags
+        env <- getSession
+        df <- liftIO $ setupDynFlagsForGHCiLike env df
+        _lp <- setSessionDynFlags df
+
+        -- copy the package state to the interactive DynFlags
+        idflags <- getInteractiveDynFlags
+        df <- getSessionDynFlags
+        setInteractiveDynFlags
+          idflags
+            { pkgState = pkgState df,
+              pkgDatabase = pkgDatabase df,
+              packageFlags = packageFlags df
+            }
+
+        -- set up a custom log action
+        setLogAction $ \_df _wr _sev _span _style _doc ->
+          defaultLogActionHPutStrDoc _df hLog _doc _style
+
+        -- load the module in the interactive environment
+        setTargets [thisModuleTarget]
+        loadResult <- load LoadAllTargets
+        case loadResult of
+          Failed -> liftIO $ do
+            hClose hLog
+            Left <$> readFile tempLog
+          Succeeded -> do
+            setContext [IIDecl (simpleImportDecl $ moduleName pRELUDE), IIModule modName]
+            Right <$> getSession
+
+    df <- liftIO $ evalGhcEnv hscEnv' getSessionDynFlags
+    let eval (stmt, l)
+          | isStmt df stmt = do
+            -- set up a custom interactive print function
+            liftIO $ writeFile temp ""
+            ctxt <- getContext
+            setContext [IIDecl (simpleImportDecl $ moduleName pRELUDE)]
+            let printFun = "let ghcideCustomShow x = Prelude.writeFile " <> show temp <> " (Prelude.show x)"
+            interactivePrint <-
+              execStmt printFun execOptions >>= \case
```

Comment:
Looks like on ghc-8.10 macOS it's silently crashing on this `execStmt`. Will try compiling with `-dynamic` to see what happens

Reply:
I develop on macOS ghc-8.10 (Nix), so I cannot explain that. 

Do you use stack or cabal? 
Does the test suite crash? 

Let me know so that I can try to repro

Reply:
This was with cabal, haven't tried with stack. I couldn't actually get it to build with dynamic because some other dependency was failing to build with it, but I have an inkling this might be due to the hombrew ghc distribution. I've run into problems with the runtime linker with it before because. I'm going to try it with the official bindist

Reply:
This was with the test suite as well by the way, forgot to mention that

Reply:
Yup looks like this is just an issue with the homebrew installed GHCs and how they don't use the in-tree integer-gmp, I wouldn't let this block this PR